app/controllers/api/irs_attempts_api_controller.rb

```diff
@@ -118,6 +118,23 @@ def authenticate_client
       end
     end
 
+    def valid_auth_token?(token)
+      salt = SecureRandom.hex(32)
+      cost = IdentityConfig.store.backup_code_cost
+
+      _digested_token = scrypt_digest(token: token, salt: salt, cost: cost)
```

Comment:
we need to name this without a leading `_` so we can use it to compare in the loop, right?

Reply:
this is kind of the tricky part. we can hash the valid tokens each time, or save the hashed versions in config so we can compare to those, or not compare any hashed versions since we'll always be storing the plaintext.

Reply:
What if we passed a proc to IdentityConfig for this obj and hashed at initialization then, so we only store the hashes in config?

Reply:
that sounds good to me!

Reply:
I gave it a shot in 40251cfded67fab1e8bac1082d2d6660e3a1f6e5, but the problem is that we can't depend on a salt from another secret using one, so I went with a plain un-salted hash for the time being 😬

Reply:
I think the primary issue with that change is we can't use a fast hash like SHA256 for this comparison, it needs to be treated more like a password and go through scrypt.

Reply:
Added 1b0bb64 which is pretty awful in a lot of ways in going outside of the conventions of how we've used the config module up to this point